consensus/polybft/checkpoint_manager.go

```diff
@@ -64,22 +69,36 @@ type checkpointManager struct {
 	logger hclog.Logger
 	// state boltDb instance
 	state *State
-	// eventGetter gets exit events (missed or current) from blocks
-	eventGetter *eventsGetter[*ExitEvent]
+	// exitEventsGetter gets exit events (missed or current) from blocks
+	exitEventsGetter *eventsGetter[*ExitEvent]
+	// slashedEventsGetter gets slashed events from blocks
+	slashedEventsGetter *eventsGetter[*contractsapi.SlashedEvent]
 }
 
 // newCheckpointManager creates a new instance of checkpointManager
 func newCheckpointManager(key ethgo.Key,
 	checkpointManagerSC types.Address, txRelayer txrelayer.TxRelayer,
 	blockchain blockchainBackend, backend polybftBackend, logger hclog.Logger,
 	state *State) *checkpointManager {
-	retry := &eventsGetter[*ExitEvent]{
+	exitEventGetter := &eventsGetter[*ExitEvent]{
 		blockchain: blockchain,
 		isValidLogFn: func(l *types.Log) bool {
 			return l.Address == contracts.L2StateSenderContract
 		},
 		parseEventFn: parseExitEvent,
 	}
+	slashedEventGetter := &eventsGetter[*contractsapi.SlashedEvent]{
```

Comment:
You can use the same event getter for both events. just set it up like `eventsGetter[contractsapi.EventAbi]`, then in  `isValidLogFn` plugin function, check if log address is from ValidatorSet or L2StateSender contracts, and on the place where you call event getter just use switch case by type to separate them into two arrays. That way we don't need to go through the same blocks to get two types of events.

Small optimization per say :smile: .

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/1808/commits/14b5cfb373f1057b108c0ed83c0f4ffe4e369289